### PR TITLE
disttask: fix removing meta when met network partition for so long then recover from it

### LINKED ISSUE #47954: add index status was always running after inject ddl owner network partition
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1、run workload
2、add index for one table of sysbench
3、add index repeatedly
4、inject ddl owner network partition

case config
      workloads:
        - type: sysbench
          storage: s3://benchmark/sysbench_64_7000w
          db: sysbench_64_7000w
          tableNum: 64
          tableSize: 70000000
          threads: 64
          subType: oltp_read_write
          ignoreErrors: "all"
      cases:
        - name: "ha_tidb(ddl-owner)_to_all_network_partition"
          warmUpTime: 1m
          faultType: network_partition
          selector: "tidb(ddl-owner)_to_all"
          period: "@every 5m"
          faultDuration: 3m
          faultTotalRunTime: 30m
          ddl:
            addIndex:
              - db: "sysbench_64_7000w"
                table: "sbtest1"
                timeout: 60m
                times: 5
                index_col_name: "c"
                job_type: "ingest"
                admin_check: false
                count_check: false

### 2. What did you expect to see? (Required)
add index can success

### 3. What did you see instead (Required)
add index status was always running after inject ddl owner network partition

### 4. What is your TiDB version? (Required)
./tidb-server -V
 Release Version: v7.6.0-alpha
Edition: Community
Git Commit Hash: cd0587eb75d70c35f01153ac5b3b6bcafaff1bee
Git Branch: heads/refs/tags/v7.6.0-alpha
UTC Build Time: 2023-10-25 11:44:53
GoVersion: go1.21.3
Race Enabled: false
Check Table Before Drop: false
Store: unistore

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: close #47954 

Problem Summary:
When a prolonged encounter network partition, the dist task framework may have chance to remove meta for the tidb node.
When the tidb node recovers from the network partition, the meta loss.
Then dispatcher may has chance to detect no available nodes to dispatch subtasks. 
### What is changed and how it works?
Add recoverMetaLoop which inits and recovers dist_framework_meta for the tidb node running the scheduler manager.
This is necessary when the TiDB node experiences a prolonged network partition and the dispatcher deletes `dist_framework_meta`.  When the TiDB node recovers from the network partition, we need to re-insert the metadata.
### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   
image

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
enhance distributed task framework when it meets prolonged network partitions.
```

### CODE REVIEW

File: pkg/disttask/framework/scheduler/manager.go [hunk 1]
```diff
@@ -138,40 +135,43 @@ func (m *Manager) Stop() {
 }
 
 // fetchAndHandleRunnableTasks fetches the runnable tasks from the global task table and handles them.
-func (m *Manager) fetchAndHandleRunnableTasks(ctx context.Context) {
+func (m *Manager) fetchAndHandleRunnableTasksLoop() {
+	defer tidbutil.Recover(metrics.LabelDomain, "fetchAndHandleRunnableTasksLoop", m.fetchAndHandleRunnableTasksLoop, false)
```
Comment (tangenta) [inline pkg/disttask/framework/scheduler/manager.go:139]:
```suggestion
	defer tidbutil.Recover(metrics.LabelDistReorg, "fetchAndHandleRunnableTasksLoop", m.fetchAndHandleRunnableTasksLoop, false)
```

Comment (ywqzzy) [reply pkg/disttask/framework/scheduler/manager.go:139]: > 

No, not only for distReorg but also import

File: pkg/disttask/framework/scheduler/manager.go [hunk 2]
```diff
@@ -97,36 +100,30 @@ func (b *ManagerBuilder) BuildManager(ctx context.Context, id string, taskTable
...
-func (m *Manager) Start() error {
-	logutil.Logger(m.logCtx).Debug("manager start")
-	var err error
+func (m *Manager) initMeta() (err error) {
 	for i := 0; i < retrySQLTimes; i++ {
 		err = m.taskTable.StartManager(m.id, config.GetGlobalConfig().Instance.TiDBServiceScope)
 		if err == nil {
 			break
 		}
-		if i%10 == 0 {
```
Comment (zimulala) [inline pkg/disttask/framework/scheduler/manager.go:109]: Why remove this？

File: pkg/disttask/framework/storage/task_table.go [hunk 3]
```diff
@@ -549,8 +549,7 @@ func (stm *TaskManager) StartSubtask(subtaskID int64) error {
 
 // StartManager insert the manager information into dist_framework_meta.
 func (stm *TaskManager) StartManager(tidbID string, role string) error {
-	_, err := stm.executeSQLWithNewSession(stm.ctx, `insert into mysql.dist_framework_meta values(%?, %?, DEFAULT)
-        on duplicate key update role = %?`, tidbID, role, role)
+	_, err := stm.executeSQLWithNewSession(stm.ctx, `replace into mysql.dist_framework_meta values(%?, %?, DEFAULT)`, tidbID, role, role)
```
Comment (zimulala) [inline pkg/disttask/framework/storage/task_table.go:552]: Is this parameter too much？

**Review by zimulala — APPROVED** [review]
LGTM
There may be some gap in expiration time with serverInfo.

**Review by tangenta — APPROVED** [review]

**Pull request merged**